lib/language_parser.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+class LanguageParser
+  def initialize(file_path_object)
+    @file = file_path_object
+  end
+
+  def messages
+    JSON.parse(YAML.load_file(file).to_json, object_class: OpenStruct)
```

Comment:
is it better to use @ for vars instead of local vars?

Reply:
no

they are different things

local variables are local variables


this is not a local variable, this is a field variable
> This means, we create a field variable inside the method...

in other words, variables with an `@` symbol are attributes of the class, that is why it gets stored once for ever.


**If this was a local variable, it wouldn't get stored forever**, because local variables only live for the duration of the method call.

I recommend you to revisit the lesson about Ruby classes in codecademy

Reply:
"is it better to use @ for vars instead of local vars?"
please read this again and tell me I don't know the difference between them

Reply:
I know there is a difference between the two.
I'm sorry, my question was confusing.

What I meant is:
Are there times were it is more efficient/less expensive to have the method run once and within the method save the returned information to the field/instance variable?

I'm guessing that depends on the circumstances, but I guess what I was really asking is:
Is this common practice or only to be done in very specific situations?

Reply:
basically store stuff in an instance variable everytime you have to **memoize the result of an expensive operation**. Not everything is going to be an expensive operation though. And not everything should stay forever with the same value.

Reply:
&#128077;